SMLHelper/Patchers/CraftTreeTypePatcher.cs

```diff
@@ -32,7 +32,7 @@ internal static ModCraftTreeRoot CreateCustomCraftTreeAndType(string name, out C
                 };
             }
 
-            if (cacheManager.IsIndexConflicting(cache.Index))
+            if (cacheManager.IsIndexConflicting(cache.Index) || cacheManager.IsIndexBanned(cache.Index))
```

Comment:
Consider combining `IsIndexConflicting` and `IsIndexBanned` into a single method since the patcher classes don't really need to worry about why the index can't be used. They just need to know it can't be so they can request another one.

Reply:
Yeah, true.

Reply:
On second thought, having to call both of this will only add to the code bloat if another enum patcher is ever added.
Best to combine these into `IsValidIndex` or `IsInvalidIndex` instead of having to remember to call both methods each time.

Reply:
To elaborate,  `IsIndexConflicting` and `IsIndexBanned` can be kept as private methods while the new, combined method calls into both of them.
Whatever the combined method is called, it should be easy to read that it is informing the caller that the index is or isn't available to use.